File: 2/02929.cpp
#include <iostream>
using namespace std;

int main() {
	string code; cin >> code;

	int index = 0, NOP = 0;
	for (int i = 0; i < code.length(); i++) {
		if (islower(code[i])) index++;
		else {
			if (index % 4 == 0) index++;
			else {
				while (index % 4) { index++; NOP++; }
				index++;
			}
		}
	}

	cout << NOP << endl;
}

File: 2/02929.py
code = input()

index = 0; NOP = 0
for i in range(len(code)):
    if code[i].islower(): index += 1
    else:
        if index % 4 == 0: index += 1
        else:
            while index % 4:
                index += 1
                NOP += 1
            index += 1

print(NOP)
